app/views/shareable_links/my_modules/step_attachments/_context_menu.html.erb

```diff
@@ -5,8 +5,15 @@
 
   <ul class="dropdown-menu dropdown-menu-right rounded !p-2.5 sn-shadow-menu-sm" aria-labelledby="dropdownAssetContextMenu">
     <li>
-      <%= link_to shared_protocol_asset_download_path(@shareable_link.uuid, asset), data: { turbolinks: false }, class: "!px-3 !py-2.5 hover:!bg-sn-super-light-blue !text-sn-blue" do %>
-        <%= t('Download') %>
+      <% download_path = if asset.step.present?
```

Comment:
merge download_step_asset and download_result_asset actions and move that check there